packages/components/combobox/src/combobox.ts

```diff
@@ -0,0 +1,807 @@
+
+    super.disconnectedCallback();
+  }
+
+  override willUpdate(changes: PropertyValues<this>): void {
+    super.willUpdate(changes);
+
+    if (changes.has('options') || changes.has('value')) {
+      this.#updateSelected();
+    }
+  }
+
+  override updated(changes: PropertyValues<this>): void {
+    super.updated(changes);
+
+    if (changes.has('autocomplete')) {
+      this.input.setAttribute('aria-autocomplete', this.autocomplete || 'both');
+    }
+
+    if (changes.has('autocomplete') || changes.has('selectOnly')) {
+      this.input.readOnly = this.selectOnly ?? this.autocomplete === 'off';
+    }
+
+    if (
+      changes.has('currentOption') ||
+      changes.has('currentSelection') ||
+      changes.has('groupSelected') ||
+      changes.has('listbox')
+    ) {
+      if (this.groupSelected && this.currentSelection.length) {
+        if (!this.#selectedGroup) {
+          this.#selectedGroup = this.shadowRoot!.createElement('sl-combobox-selected-group') as SelectedGroup;
+          this.#selectedGroup.addEventListener('click', this.#onOptionsClick);
+        }
+        this.#selectedGroup.currentOption = this.currentOption;
+        this.#selectedGroup.hasGroups = !!this.listbox?.querySelector('sl-option-group');
+        this.#selectedGroup.options = this.currentSelection;
+
+        if (this.#selectedGroup.parentElement !== this.listbox) {
+          this.listbox?.prepend(this.#selectedGroup);
+        }
+      } else {
+        this.#selectedGroup?.remove();
+        this.#selectedGroup = undefined;
+      }
+    }
+
+    if (changes.has('disabled')) {
+      this.input.disabled = !!this.disabled;
+    }
+
+    if (changes.has('required')) {
+      this.internals.ariaRequired = this.required ? 'true' : 'false';
+
+      this.#updateValidity();
+    }
+  }
+
+  override render(): TemplateResult {
+    return html`
+      <sl-text-field
+        @input=${this.#onInput}
+        @keydown=${this.#onKeydown}
+        @sl-blur=${this.#onTextFieldBlur}
+        @sl-change=${this.#onTextFieldChange}
+        @sl-focus=${this.#onTextFieldFocus}
+        @sl-update-state=${this.#onTextFieldUpdateState}
+        ?disabled=${this.disabled}
+        ?readonly=${this.selectOnly}
+        ?required=${this.required}
+        placeholder=${ifDefined(this.placeholder)}
+        size=${ifDefined(this.size)}
+      >
+        ${this.multiple && this.currentSelection.length
+          ? html`
+              <sl-tag-list
+                size=${ifDefined(this.size)}
+                slot="prefix"
+                stacked
+                .emphasis=${this.disabled ? 'bold' : 'subtle'}
+              >
+                ${repeat(
+                  this.currentSelection,
+                  option => option.id,
+                  option => html`
+                    <sl-tag
+                      @sl-remove=${() => this.#onRemove(option)}
+                      ?disabled=${this.disabled}
+                      ?removable=${!this.disabled}
```

Comment:
Are we sure we want to always have removable tags when it's not disabled? Maybe we want to have a variant with tags inside that we cannot remove?

Reply:
I think that is outside of the scope of the first version.